test/integration/smoke/test_multiple_public_ip_ranges.py

```diff
@@ -0,0 +1,887 @@
+            domainid=cls.domain.id
+        )
+        cls.services["publiciprange"]["zoneid"] = cls.zone.id
+        cls.service_offering = ServiceOffering.create(
+            cls.apiclient,
+            cls.services["service_offerings"]["tiny"]
+        )
+        cls.virtual_machine = VirtualMachine.create(
+            cls.apiclient,
+            cls.services["virtual_machine"],
+            templateid=template.id,
+            accountid=cls.account.name,
+            domainid=cls.account.domainid,
+            serviceofferingid=cls.service_offering.id
+        )
+        cls.defaultNetworkId = cls.virtual_machine.nic[0].networkid
+        cls._cleanup = [
+            cls.virtual_machine,
+            cls.account,
+            cls.service_offering
+        ]
+
+    def setUp(self):
+        self.apiclient = self.testClient.getApiClient()
+        self.cleanup = []
+        return
+
+    @classmethod
+    def tearDownClass(cls):
+        try:
+            cls.apiclient = super(
+                TestStaticNat,
+                cls).getClsTestClient().getApiClient()
+            cleanup_resources(cls.apiclient, cls._cleanup)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+
+    def tearDown(self):
+        cleanup_resources(self.apiclient, self.cleanup)
+        return
+
+    @attr(tags=["advanced", "smoke"], required_hardware="true")
+    def test_static_nat_on_ip_from_non_src_nat_ip_range(self):
+        """Test for static nat on a IP which is in pubic IP range different
+           from public IP range that has source NAT IP associated with network
+        """
+
+        # Validate the following:
+        # 1. Create a new public IP range and dedicate to a account
+        # 2. Acquire a IP from new public range
+        # 3. Enable static NAT on acquired IP from new range
+        # 4. Create a firewall rule to open up the port
+        # 5. Test SSH works to the VM
+
+        self.public_ip_range = PublicIpRange.create(
+                                    self.apiclient,
+                                    self.services["publiciprange"]
+                               )
+
+        self.debug("Dedicating Public IP range to the account");
+        dedicate_public_ip_range_response = PublicIpRange.dedicate(
+                                                self.apiclient,
+                                                self.public_ip_range.vlan.id,
+                                                account=self.account.name,
+                                                domainid=self.account.domainid
+                                            )
+        ip_address = PublicIPAddress.create(
+            self.apiclient,
+            self.account.name,
+            self.zone.id,
+            self.account.domainid,
+            self.services["virtual_machine"]
+        )
+        self.cleanup.append(ip_address)
+        self.cleanup.append(self.public_ip_range)
```

Comment:
Please see previous notes regarding extracting the dedication/creation of public IP ranges to private utility method.